### DISCUSSION TITLE
Adding custom flags or extending native flags

### DISCUSSION THREAD

**Discussion by proxemy** [wintermeyer/animina#970] [Ideas]
There are certainly personal traits missing in the predefined selection of flags, that instantly qualify as red/green flags for some individuals. In my case, it would be the Hobby Shopping but there are surely many more.

The difficulty would be to generalize all custom flags to extend and auto complete the selection UI and with that, adjusting the matching algorithm.

I have not looked at the code but since you mention a functional paradigm, you might run into headaches when faced with this new non-algebraic user input meant for the matching algorithm, so as a compromise, maybe just add functionality to suggest missing flags, to manually add them to the curated list of native flags.

**Comment by wintermeyer** [comment]
I do like the `shopping` flag idea. 😉

Thanks for your feedback @proxemy! The whole flag system is powerful but I do realise its limitations too. You have no idea how much time went into the current flags. So much is still missing and so much is not needed by everybody at the same time.

Let me think about this a bit more.

**Comment by wintermeyer** [comment]
Let's make `shopping` happen: https://github.com/animina-dating/animina/issues/971
